### PR TITLE
PullRequestDiscussion entity

### PR DESCRIPTION
**Pull request by xtuchyna**
## Related Issues and Dependencies
Implements #301 

## This introduces a breaking change

- [ ] Yes
- [X] No

## Description
New `PullRequestDiscussion` entity introduced.
Attributes:
 - commenter login
 - comment text
 - comment created_date
 - comment reactions (in form of `:emoji:`)
A quick peak on the knowledge structure (mi-scheduler repo analysis):
```
...
    {
      "user": "todo[bot]",
      "text": "## Rename in common to mi\n\nhttps://github.com/thoth-station/mi-scheduler/blob/bb23b67dda817352794bfaf7d075e1f6ca245496/app.py#L93-L97\n\n---\n\n###### This comment was generated by [todo](https://todo.jasonet.co) based on a `TODO` comment in bb23b67dda817352794bfaf7d075e1f6ca245496 in #38. cc @xtuchyna.",
      "created_at": 1596022927,
      "reactions": []
    },
...
```

### CODE REVIEW

File: srcopsmetrics/entities/pull_request_discussion.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+from github.PaginatedList import PaginatedList
+from github.PullRequest import PullRequest
+from voluptuous.schema_builder import Schema
+
+from srcopsmetrics.entities import Entity
+
+
+class PullRequestDiscussion(Entity):
+    """Template entity.
```
Comment (pacospace) [inline srcopsmetrics/entities/pull_request_discussion.py:29]: don't we need to customize all docstring after using the template?

File: Pipfile [hunk 2]
```diff
@@ -3,13 +3,16 @@ url = "https://pypi.python.org/simple"
...
+
 [packages]
 numpy = "*"
 pygithub = "*"
 matplotlib = "*"
 pandas = "*"
 thoth-storages = "*"
 semver = "*"
+idna_ssl = "*"
+typing_extensions = "*"
```
Comment (pacospace) [inline Pipfile:15]: Are these required direct dependencies?

Comment (xtuchyna) [reply Pipfile:15]: This was a problem with `pytest`, but I still don't know why.
```
======================================================================
ERROR: bot_knowledge (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: bot_knowledge
Traceback (most recent call last):
  File "/usr/lib64/python3.6/unittest/loader.py", line 153, in loadTestsFromName
    module = __import__(module_name)
  File "/workspace/repo/srcopsmetrics/bot_knowledge.py", line 27, in <module>
    from srcopsmetrics.entities import Entity, NOT_FOR_INSPECTION
  File "/workspace/repo/srcopsmetrics/entities/__init__.py", line 19, in <module>
    from .interface import Entity
  File "/workspace/repo/srcopsmetrics/entities/interface.py", line 32, in <module>
    from srcopsmetrics.entities.tools.storage import KnowledgeStorage
  File "/workspace/repo/srcopsmetrics/entities/tools/storage.py", line 26, in <module>
    from thoth.storages.ceph import CephStore
  File "/opt/app-root/lib/python3.6/site-packages/thoth/storages/__init__.py", line 30, in <module>
    from .graph import GraphDatabase
  File "/opt/app-root/lib/python3.6/site-packages/thoth/storages/graph/__init__.py", line 21, in <module>
    from .postgres import GraphDatabase
  File "/opt/app-root/lib/python3.6/site-packages/thoth/storages/graph/postgres.py", line 53, in <module>
    from thoth.python import PackageVersion
  File "/opt/app-root/lib/python3.6/site-packages/thoth/python/__init__.py", line 29, in <module>
    from .aiosource import AIOSource, AsyncIterablePackages, AsyncIterableVersions, AsyncIterableArtifacts
  File "/opt/app-root/lib/python3.6/site-packages/thoth/python/aiosource.py", line 29, in <module>
    import aiohttp
  File "/opt/app-root/lib/python3.6/site-packages/aiohttp/__init__.py", line 6, in <module>
    from .client import (
  File "/opt/app-root/lib/python3.6/site-packages/aiohttp/client.py", line 35, in <module>
    from . import hdrs, http, payload
  File "/opt/app-root/lib/python3.6/site-packages/aiohttp/http.py", line 7, in <module>
    from .http_parser import (
  File "/opt/app-root/lib/python3.6/site-packages/aiohttp/http_parser.py", line 15, in <module>
    from .helpers import NO_EXTENSIONS, BaseTimerContext
  File "/opt/app-root/lib/python3.6/site-packages/aiohttp/helpers.py", line 62, in <module>
    import idna_ssl
ModuleNotFoundError: No module named 'idna_ssl'
```

The typing_extensions was a similar error, catched by local `pytest` check:

```
======================================================================
ERROR: iterator (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: iterator
Traceback (most recent call last):
  File "/usr/lib64/python3.8/unittest/loader.py", line 154, in loadTestsFromName
    module = __import__(module_name)
  File "/home/xtuchyna/git/forks/SrcOpsMetrics/srcopsmetrics/iterator.py", line 30, in <module>
    from srcopsmetrics.entities import Entity
  File "/home/xtuchyna/git/forks/SrcOpsMetrics/srcopsmetrics/entities/__init__.py", line 19, in <module>
    from .interface import Entity
  File "/home/xtuchyna/git/forks/SrcOpsMetrics/srcopsmetrics/entities/interface.py", line 32, in <module>
    from srcopsmetrics.entities.tools.storage import KnowledgeStorage
  File "/home/xtuchyna/git/forks/SrcOpsMetrics/srcopsmetrics/entities/tools/storage.py", line 26, in <module>
    from thoth.storages.ceph import CephStore
  File "/home/xtuchyna/.local/share/virtualenvs/SrcOpsMetrics-Vwmk3Tpf/lib/python3.8/site-packages/thoth/storages/__init__.py", line 30, in <module>
    from .graph import GraphDatabase
  File "/home/xtuchyna/.local/share/virtualenvs/SrcOpsMetrics-Vwmk3Tpf/lib/python3.8/site-packages/thoth/storages/graph/__init__.py", line 21, in <module>
    from .postgres import GraphDatabase
  File "/home/xtuchyna/.local/share/virtualenvs/SrcOpsMetrics-Vwmk3Tpf/lib/python3.8/site-packages/thoth/storages/graph/postgres.py", line 53, in <module>
    from thoth.python import PackageVersion
  File "/home/xtuchyna/.local/share/virtualenvs/SrcOpsMetrics-Vwmk3Tpf/lib/python3.8/site-packages/thoth/python/__init__.py", line 29, in <module>
    from .aiosource import AIOSource, AsyncIterablePackages, AsyncIterableVersions, AsyncIterableArtifacts
  File "/home/xtuchyna/.local/share/virtualenvs/SrcOpsMetrics-Vwmk3Tpf/lib/python3.8/site-packages/thoth/python/aiosource.py", line 26, in <module>
    import aiohttp
  File "/home/xtuchyna/.local/share/virtualenvs/SrcOpsMetrics-Vwmk3Tpf/lib64/python3.8/site-packages/aiohttp/__init__.py", line 6, in <module>
    from .client import (
  File "/home/xtuchyna/.local/share/virtualenvs/SrcOpsMetrics-Vwmk3Tpf/lib64/python3.8/site-packages/aiohttp/client.py", line 35, in <module>
    from . import hdrs, http, payload
  File "/home/xtuchyna/.local/share/virtualenvs/SrcOpsMetrics-Vwmk3Tpf/lib64/python3.8/site-packages/aiohttp/http.py", line 7, in <module>
    from .http_parser import (
  File "/home/xtuchyna/.local/share/virtualenvs/SrcOpsMetrics-Vwmk3Tpf/lib64/python3.8/site-packages/aiohttp/http_parser.py", line 15, in <module>
    from .helpers import NO_EXTENSIONS, BaseTimerContext
  File "/home/xtuchyna/.local/share/virtualenvs/SrcOpsMetrics-Vwmk3Tpf/lib64/python3.8/site-packages/aiohttp/helpers.py", line 48, in <module>
    from typing_extensions import Protocol
ModuleNotFoundError: No module named 'typing_extensions'
```
After adding them, these issues were fixed.

Comment (xtuchyna) [reply Pipfile:15]: it seems that this happend after `thoth-storages` update, look here #299

Comment (pacospace) [reply Pipfile:15]: Did you try deleting and recreating Pipfile.lock but without adding those two to direct dependencies?

Comment (xtuchyna) [reply Pipfile]: @pacospace problem solved :beers:

**Review by pacospace — APPROVED** [review]
lgtm 💯

**Pull request merged**